src/gc.h

```diff
@@ -449,6 +449,37 @@ extern int gc_n_threads;
 extern jl_ptls_t* gc_all_tls_states;
 extern gc_heapstatus_t gc_heap_stats;
 
+STATIC_INLINE int gc_first_parallel_collector_thread_id(void) JL_NOTSAFEPOINT
+{
+    return gc_first_tid;
```

Comment:
What is this when `jl_n_markthreads == 0`?

Reply:
This function should not be called when `jl_markthreads = 0` to begin with.

There should probably be an assertion there to abort when `jl_markthreads = 0`.

Reply:
Edited my previous comment: should probably just return 0. We just want it to return a value which is greater than last_thread_id (-1 when `jl_n_markthreads == 0`) so that lower-bounded by first and upper-bounded by last never runs.